rts/lra/lra-test/src/main/java/io/narayana/lra/participant/api/ActivityController.java

```diff
@@ -560,8 +560,25 @@ private void endCheck(Activity activity) {
         activity.setHow(null);
         activity.setArg(null);
 
-        if ("wait".equals(how) && arg != null && "recovery".equals(arg)) {
-            lraClient.getRecoveringLRAs();
+        if ("wait".equals(how)) {
+            if (arg != null) {
+                if ("recovery".equals(arg)) {
+                    lraClient.getRecoveringLRAs(); // run a recovery scan
+                } else {
+                    int ms = 0;
+
+                    try {
+                        ms = Integer.getInteger(arg, 0);
+                    } catch (Exception ignore) {
+                    }
+
+                    try {
+                        Thread.sleep(ms <= 0 ? Long.MAX_VALUE : ms); // delay the end call
```

Comment:
I would like to understand the rationalization of giving to user a way of delaying end call?
It's seems as some workaround to something. I can be a wrong in understanding but letting the client to drive timing from outside sounds to me supportive of race conditions.
I know we discussed some timeouts to be involved but I can't connect this with the discussion.

Reply:
ouch, I see. Even I was checking the package I missed that. I'm sorry,  this is fine.

Reply:
Notice that I also added a comment to indicate what the recovery scan is testing here.